tests/options/pugClassLocation/pug-class-location.test.ts

```diff
@@ -0,0 +1,56 @@
+/* eslint-disable quotes */
+import { format } from 'prettier';
+import { plugin } from './../../../src/index';
+
+describe('Options', () => {
+	describe('pugClassLocation', () => {
+		test('should position class literals before attributes', () => {
+			const expected: string = `.before-class.class.class-before(type="button", class=boolean ? "test" : "fact")= "type"\n`;
+			const code: string = `div.before-class(type="button", class=boolean ? "test" : "fact").class.class-before= "type"`;
+			const actual: string = format(code, {
+				parser: 'pug',
+				plugins: [plugin],
+
+				pugClassLocation: 'before-attributes'
+			});
+
+			expect(actual).toBe(expected);
+		});
+		test('should position class literals after attributes', () => {
+			const expected: string = `(type="button", class=boolean ? "test" : "fact").class.class-before.after-class= "type"\n`;
```

Comment:
:eyes: does it really remove the `div` tag?
This doesn't look valid.

Reply:
it does as the default.

Reply:
Using e.g. https://pughtml.com it gives me `unexpected token "start-attributes" - line 1:1`
so... this would be a bug :lady_beetle:

Reply:
this is actually general bug.
`(attr="test for no id or class") text` won't be reformatted correctly when `pugExplicitDiv` is set to `true` or `false`

Reply:
not sure that is something prettier should since it is just malformed pug anyways.

Reply:
I think you misunderstood something. I mean expected should be `const expected: string = `div(type="button", ...`
The test expectation is wrong here. A test should expect the correct thing not just the response it currently gives.
Then it fails and we need to fix the code until it response the correct result.

Reply:
I understood that. 
I'm saying that if the string contains `(attr="test for no id or class") text` it will not add the `div`.

Reply:
If it would be the `actual`, yes it would be invalid, but this line is the `expected`

```suggestion
			const expected: string = `div(type="button", class=boolean ? "test" : "fact").class.class-before.after-class= "type"\n`;
```

Reply:
I'm talking about in general, not this test example.

Reply:
pugExplicitDiv false only removes it if it's possible to omit it. so e.g. when using `.my-class(someattr="anything")` this is valid pug, with explicit div it would be `div.my-class(someattr="anything")`

Reply:
yes but if the line is incorrect to begin with it won't add the div.

Reply:
sure, because this is already false pug and therefore would not trigger my plugin printer at all